### PR TITLE
Fix PHP Fatal Error: Nesting level too deep - recursive dependency.

### PR DESCRIPTION
**Pull request by kf99916**
Fix PHP Fatal Error: Nesting level too deep - recursive dependency.
The reference of this solution is from the issue of Yii2: https://github.com/yiisoft/yii2/issues/8348

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-	"name": "yiisoft/yii",
+	"name": "kf99916/yii",
```
Comment (tomtomsen) [inline composer.json]: i dont think thats a good change. the project shouldn't be renamed

Comment (samdark) [inline composer.json]: Definitely not :(

Comment (kf99916) [inline composer.json]: I'm sorry about that. Please ignore changes in composer.json. My project is dependent on Yii 1.1.17, however, there is a issue about sorting of CArrayDataProvider. Reviewing pull request and releasing next version needs time. In order to ensure my project can move on the next version smoothly, so I let my project be dependent on my repo.

**Comment by kf99916** [comment]
@samdark and @tomtomsen , I reverted the composer change. Could you help review the pull request again? Thank you so much. :)

**Comment by samdark** [comment]
It would be great to have a test for it like https://github.com/yiisoft/yii2/commit/0f30b0843b02d6f163e62c626dffd0299a0580d2

**Comment by kf99916** [comment]
UT for sorting nested objects is added and format is also aligned.

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii/commit/9ffe1da7a94d78996a4e02bbce3c449d06240907. Thanks!

**Pull request closed without merging**